include/uart.h

```diff
@@ -64,6 +65,62 @@ extern "C" {
  */
 #define UART_ERROR_BREAK    (1 << 3)
 
+/**
+ * @brief UART controller configuration structure
+ *
+ * @param baudrate   Baudrate setting in bps.
+ * @param rx_timeout RX timeout in ms.
+ * @param flags      Bit field with following parts:
+ *
+ *     parity       [ 0 : 2 ]    - Parity bits.
+ *     stop_bits    [ 3 : 4 ]    - Stop bits.
+ *     data_bits    [ 5 : 7 ]    - Data bits.
+ *     flow_ctrl    [ 8 : 9 ]    - Flow control setting.
+ */
+struct uart_config {
+	u32_t baudrate;
+	u32_t rx_timeout;
```

Comment:
Does rx_timeout have any particular meaning (might be held over from other PR)?

Reply:
Actually, +1 to what @galak says - let's leave rx_timeout to the other PR/part of the API.

Reply:
Right, removed rx_timeout